### ISSUE TITLE
False positive for `ruff` `I001` (isort) when there is a function/class definition after the imports cell

### ISSUE THREAD

**Issue by felix-cw** [nbQA-dev/nbQA#806]
This bug is similar to #796 (thank you very much for fixing that!).
It arises when running `nbqa ruff`

- on a notebook with any import in the first cell
- where the isort checks are enabled e.g. with --select=I
- The cell following the cell containing imports starts with a class or function definition

As in #796, there is a false positive error message of 

```
I001 [*] Import block is un-sorted or un-formatted
```

and `--fix` does not change the notebook, even though it reports it as fixed.

My minimal notebook has the following cells:

```python
import os
```

```python
class A:
    ...
```

If I add the `--diff` option, it wants to insert a newline

```diff
@@ -1,6 +1,7 @@
 # %%NBQA-CELL-SEPfc4e93
 import os

+
 # %%NBQA-CELL-SEPfc4e93
 class A:
     ...

Would fix 1 error.
```

**Comment by MarcoGorelli** [comment]
thanks @felix-cw for the report, will take a look

**Comment by MarcoGorelli** [comment]
Right, so `ruff` wants two newlines before a definition or a class, and one newline otherwise, it seems:
```python
# %%NBQA-CELL-SEPb4b132
import os

# %%NBQA-CELL-SEPb4b132
cwd = os.getcwd()
x = np.arange(1, 10)

# %%NBQA-CELL-SEPb4b132
import foo


# %%NBQA-CELL-SEPb4b132
class Foo:
    ...

# %%NBQA-CELL-SEPb4b132
import foo


# %%NBQA-CELL-SEPb4b132
def foo():
    ...
```
The following passes

I may need to preprocess the temporary Python file before passing it to `ruff`...

**Comment by MarcoGorelli** [comment]
one solution could be to pass autopep8 with only codes E3 selected first

this'd be a pretty big change, and I'll need to test it out thoroughly first. it might even warrant a 2.0 release. but it might be the right thing to do

**Comment by MarcoGorelli** [comment]
@felix-cw this should be solved in version 1.7.0, would appreciate it if you could check - thanks 🙏

**Comment by MarcoGorelli** [comment]
of course! @all-contributors please add @felix-cw  for infra
